### ISSUE TITLE
Research: Integrated Force Regime Orchestration Study

### ISSUE THREAD

**Issue by fermga** [fermga/TNFR-Python-Engine#2929]
# Integrated Force Regime Orchestration Study

## Context

Three structural fields (|∇φ|, K_φ, ξ_C) have been individually investigated with weak-to-moderate correlations. The fourth field (Φ_s) achieved CANONICAL status with strong correlation (corr = -0.822). This integration study investigates whether **combined field dynamics** exhibit emergent predictive power exceeding individual components.

**Current Status**: All fields RESEARCH except Φ_s (CANONICAL)

**Individual Correlations** (from §10-14):
- Φ_s (structural potential): corr = -0.822 (CANONICAL)
- |∇φ| (phase gradient): corr ≈ -0.13 (EM-like)
- K_φ (phase curvature): corr ≈ -0.07 (strong-like)
- ξ_C (coherence length): Divergence at I_c = 2.015 (weak-like)

## Objective

Investigate whether the four structural fields exhibit **orchestrated dynamics** analogous to fundamental force unification, and develop **composite metrics** with predictive power exceeding individual fields.

### 1. Field Interaction Matrix

**Goal**: Quantify cross-correlations between all four fields

**Implementation**:
```python
def compute_field_interaction_matrix(experiments):
    """
    Compute pairwise correlations and partial correlations.
    
    Fields: Φ_s, |∇φ|, K_φ, ξ_C
    """
    fields = {
        'phi_s': [exp['phi_s_drift'] for exp in experiments],
        'grad_phi': [exp['grad_phi_mean'] for exp in experiments],
        'k_phi': [exp['k_phi_max'] for exp in experiments],
        'xi_c': [exp['xi_c'] for exp in experiments],
        'delta_c': [exp['delta_c'] for exp in experiments]
    }
    
    # Pearson correlation matrix
    corr_matrix = np.corrcoef([fields[k] for k in fields.keys()])
    
    # Partial correlations (controlling for Φ_s)
    from sklearn.covariance import GraphicalLassoCV
    estimator = GraphicalLassoCV()
    estimator.fit(np.column_stack([fields[k] for k in fields.keys()]))
    partial_corr = estimator.precision_  # Inverse covariance
    
    return corr_matrix, partial_corr
```

**Experiments**:
- [ ] Compute correlation matrix across all 2,400+ experiments
- [ ] Identify dominant field pairs (|corr| > 0.3)
- [ ] Compute partial correlations controlling for Φ_s (does |∇φ| or K_φ add information?)
- [ ] Test for multicollinearity (VIF analysis)

**Expected Findings**:
- **Φ_s dominant**: Highest correlations with ΔC
- **|∇φ| and K_φ independent**: Low cross-correlation (orthogonal aspects)
- **ξ_C conditional**: Strong correlation near I_c, weak elsewhere

**Acceptance Criteria**:
- Full correlation matrix documented
- Partial correlations show |∇φ|/K_φ add ≥5% explained variance beyond Φ_s

### 2. Composite Field Metrics

**Goal**: Develop weighted combinations of fields with higher predictive power

**Approach 1: Linear Combination**
```python
def composite_field_score(phi_s, grad_phi, k_phi, xi_c, weights):
    """
    Weighted linear combination of normalized fields.
    
    Score = w1*Φ_s + w2*|∇φ| + w3*K_φ + w4/ξ_C
    """
    # Normalize each field to [0, 1]
    phi_s_norm = (phi_s - phi_s_min) / (phi_s_max - phi_s_min)
    # ... (normalize others)
    
    score = (weights[0] * phi_s_norm + 
             weights[1] * grad_phi_norm + 
             weights[2] * k_phi_norm + 
             weights[3] / xi_c_norm)  # Inverse: high ξ_C = stable
    return score
```

**Approach 2: Principal Component Analysis**
```python
from sklearn.decomposition import PCA

def extract_principal_field(experiments):
    """
    Find linear combination explaining maximum variance.
    """
    X = np.column_stack([
        experiments['phi_s_drift'],
        experiments['grad_phi_mean'],
        experiments['k_phi_max'],
        1 / experiments['xi_c']  # Inverse for stability direction
    ])
    
    pca = PCA(n_components=1)
    pc1 = pca.fit_transform(X)
    loadings = pca.components_[0]
    
    return pc1, loadings
```

**Approach 3: Machine Learning Ensemble**
```python
from sklearn.ensemble import RandomForestRegressor

def ml_composite_predictor(X_train, y_train):
    """
    Train ensemble model: Fields → ΔC prediction.
    """
    features = ['phi_s', 'grad_phi', 'k_phi', 'xi_c']
    model = RandomForestRegressor(n_estimators=100, max_depth=5)
    model.fit(X_train[features], y_train['delta_c'])
    
    # Feature importance
    importances = dict(zip(features, model.feature_importances_))
    return model, importances
```

**Experiments**:
- [ ] Grid search optimal weights for linear combination (maximize |corr| with ΔC)
- [ ] PCA: Determine PC1 loadings and variance explained
- [ ] Train RF model: Compare R² with Φ_s alone (R² = 0.68)
- [ ] Cross-validation: Test generalization to held-out topologies

**Target**: Composite metric with |corr| > 0.85 or R² > 0.72 (improvement over Φ_s alone)

**Acceptance Criteria**:
- Linear combination achieves |corr| ≥ 0.83 (vs 0.82 for Φ_s)
- PCA PC1 explains ≥ 75% variance across fields
- ML model R² ≥ 0.72 on test set

### 3. Force Regime Phase Diagram

**Goal**: Map intensity-topology space to dominant field regimes

**Theory**: Different conditions may activate different "force-like" mechanisms:
- **Low I**: Φ_s dominates (gravity-like, stable equilibrium)
- **Medium I**: |∇φ| significant (EM-like, long-range stress)
- **High I near I_c**: K_φ and ξ_C critical (strong/weak-like, confinement + breaking)

**Implementation**:
```python
def classify_force_regime(phi_s_drift, grad_phi, k_phi, xi_c):
    """
    Classify which field is dominant.
    
    Criteria:
    - Gravity-like: Φ_s_drift > 2.0 (escape threshold)
    - EM-like: |∇φ| > 1.5 (high gradient)
    - Strong-like: K_φ > 4.8 (confinement threshold)
    - Weak-like: ξ_C diverging (dξ/dI < -50)
    """
    regimes = []
    if phi_s_drift > 2.0:
        regimes.append('gravity')
    if grad_phi > 1.5:
        regimes.append('EM')
    if k_phi > 4.8:
        regimes.append('strong')
    if xi_c > 200:  # Divergence proxy
        regimes.append('weak')
    
    # Dominant regime: most critical threshold exceeded
    return regimes if regimes else ['stable']
```

**Experiments**:
- [ ] Classify all 2,400+ experiments by dominant regime
- [ ] Create 2D phase diagram: intensity vs topology, colored by regime
- [ ] Test regime transitions: Do sequences cross regime boundaries?
- [ ] Measure regime mixing: Can multiple forces coexist?

**Expected Patterns**:
- **Stable zone** (I < 2.0): Gravity-only (low Φ_s, low gradients)
- **Critical zone** (I ≈ 2.0-2.1): All four forces active (unification regime)
- **Chaotic zone** (I > 2.1): EM + Strong dominant (high gradients + curvature)

**Acceptance Criteria**:
- Phase diagram with ≥3 distinct zones
- Regime transitions align with known thresholds (I_c = 2.015, Φ_s < 2.0, K_φ < 4.88)
- Predictive: Regime classification improves fragmentation prediction (AUC > 0.9)

### 4. Temporal Orchestration Analysis

**Goal**: Identify if fields exhibit sequential activation (force hierarchy)

**Theory**: Fundamental forces emerged sequentially in early universe:
1. Gravity (first, long-range)
2. Strong + EM (later, as universe cooled)
3. Weak (last, symmetry breaking)

**TNFR Hypothesis**: Fields activate in sequence during destabilization:
1. Φ_s increases first (global drift from equilibrium)
2. |∇φ| rises second (directional stress propagates)
3. K_φ spikes third (local torsion accumulates)
4. ξ_C drops last (fragmentation completes)

**Implementation**:
```python
def analyze_temporal_sequence(trajectory):
    """
    Track field evolution over time.
    
    trajectory: List of graph states [G_0, G_1, ..., G_T]
    """
    time_series = {
        'phi_s': [],
        'grad_phi': [],
        'k_phi': [],
        'xi_c': [],
        'coherence': []
    }
    
    for G_t in trajectory:
        phi_s = np.mean(list(compute_structural_potential(G_t).values()))
        grad_phi = np.mean(list(compute_phase_gradient(G_t).values()))
        k_phi = np.max([abs(k) for k in compute_phase_curvature(G_t).values()])
        xi_c = estimate_coherence_length(G_t)
        c_t = compute_coherence(G_t)
        
        time_series['phi_s'].append(phi_s)
        time_series['grad_phi'].append(grad_phi)
        time_series['k_phi'].append(k_phi)
        time_series['xi_c'].append(xi_c)
        time_series['coherence'].append(c_t)
    
    # Detect peaks and order
    peak_times = {}
    for field in ['phi_s', 'grad_phi', 'k_phi']:
        peak_times[field] = np.argmax(time_series[field])
    peak_times['xi_c'] = np.argmin(time_series['xi_c'])  # Minimum = collapse
    
    return time_series, peak_times
```

**Experiments**:
- [ ] Simulate 100+ trajectories approaching fragmentation
- [ ] Track field evolution at each step
- [ ] Measure average peak ordering across trajectories
- [ ] Test if valid sequences show different ordering than violations

**Expected Sequence** (violation trajectories):
1. Φ_s peak at step t
2. |∇φ| peak at step t+1 to t+3
3. K_φ peak at step t+2 to t+4
4. ξ_C minimum at step t+3 to t+5

**Acceptance Criteria**:
- Consistent ordering in ≥70% of fragmentation trajectories
- Lead times: Φ_s alerts 3-5 steps before fragmentation, |∇φ| 2-4 steps, K_φ 1-2 steps
- Valid sequences abort sequence before K_φ spike (IL intervention)

### 5. Operator-Field Coupling Analysis

**Goal**: Map operators to field responses (which operators affect which fields?)

**Theory**: Operators should have characteristic field signatures:
- **OZ (dissonance)**: Increases all fields (ΔΦ_s ↑, |∇φ| ↑, K_φ ↑)
- **IL (coherence)**: Decreases all fields (ΔΦ_s ↓, |∇φ| ↓, K_φ ↓)
- **RA (resonance)**: Decreases |∇φ| (phase alignment), may increase ξ_C (range extension)
- **ZHIR (mutation)**: Spikes K_φ locally (torsion), may reset Φ_s (new equilibrium)

**Implementation**:
```python
def operator_field_signature(G_before, G_after, operator_type):
    """
    Measure field changes induced by operator.
    """
    fields_before = {
        'phi_s': np.mean(list(compute_structural_potential(G_before).values())),
        'grad_phi': np.mean(list(compute_phase_gradient(G_before).values())),
        'k_phi': np.max([abs(k) for k in compute_phase_curvature(G_before).values()]),
        'xi_c': estimate_coherence_length(G_before)
    }
    
    fields_after = {k: ... for k in fields_before.keys()}  # Compute after
    
    delta_fields = {k: fields_after[k] - fields_before[k] for k in fields_before.keys()}
    
    return delta_fields
```

**Experiments**:
- [ ] Apply each operator type (AL, EN, IL, OZ, UM, RA, ZHIR, etc.) individually
- [ ] Measure ΔΦ_s, Δ|∇φ|, ΔK_φ, Δξ_C for N=50+ applications
- [ ] Create operator-field response matrix (13 operators × 4 fields)
- [ ] Identify characteristic signatures (e.g., RA uniquely extends ξ_C)

**Acceptance Criteria**:
- Signatures significantly different (ANOVA p < 0.01) for ≥8/13 operators
- RA shows Δξ_C > 0 while OZ shows Δξ_C < 0 (opposing effects)
- IL consistently reduces all instability fields (ΔK_φ < 0, Δ|∇φ| < 0)

### 6. Cross-Domain Unification Test

**Goal**: Test if field orchestration patterns replicate across biological/social/AI domains

**Theory**: If TNFR fields truly capture universal interaction dynamics, force regime patterns should emerge in:
- **Biological**: Neural avalanches (criticality), cardiac arrhythmia (fragmentation)
- **Social**: Opinion polarization (symmetry breaking), consensus formation (phase alignment)
- **AI**: Training dynamics (loss landscape = Φ_s), mode collapse (fragmentation)

**Implementation**:
- [ ] Biological: Neural oscillator network (Kuramoto model adapted)
- [ ] Social: Voter model with opinion phases
- [ ] AI: Distributed SGD with gradient phases

**Experiments**:
- [ ] Simulate domain-specific dynamics
- [ ] Compute all four fields using domain-adapted definitions
- [ ] Test if critical transitions show orchestrated field activation
- [ ] Compare field correlation patterns with TNFR baseline

**Acceptance Criteria**:
- All four fields identifiable in ≥2 domains
- Critical transitions (avalanche, polarization, mode collapse) show Φ_s + |∇φ| + K_φ + ξ_C activation
- Correlation patterns within 30% of TNFR values

## Promotion Criteria to Canonical (Integrated)

**Individual fields remain RESEARCH unless they individually meet criteria**. This study investigates:

1. **Composite Metrics**: Can combined fields achieve |corr| > 0.85 or R² > 0.75?
2. **Regime Classification**: Does force regime improve fragmentation prediction (AUC > 0.95)?
3. **Universal Orchestration**: Do field sequences replicate across ≥2 domains?

**If successful**: Establish **TNFR Force Dynamics Framework** as canonical analysis layer (even if individual fields remain research telemetry)

## Constraints

**Preserve Invariants**:
- **Invariant #1**: No direct EPI mutation (all fields read-only)
- **Invariant #10**: Domain neutrality in core; force analogies isolated to analysis

**Classification**: Integrated framework RESEARCH pending validation

## Tools & References

**Existing Tools**:
- `src/tnfr/physics/fields.py` (all four field computations)
- Individual field analysis tools (from Issues #2926, #2927, #2928)

**New Tools Needed**:
- `tools/field_interaction_matrix.py`
- `tools/composite_field_optimizer.py`
- `tools/force_regime_classifier.py`
- `tools/temporal_orchestration_tracker.py`
- `tools/operator_field_signature_analyzer.py`

**Documentation**:
- `docs/TNFR_FORCES_EMERGENCE.md` (§1-15 complete evidence)
- `AGENTS.md` § Structural Fields

## Expected Outcomes

**Best Case**:
- Composite metric with |corr| = 0.90 (vs 0.82 for Φ_s alone)
- Force regime classification AUC = 0.98 (near-perfect fragmentation prediction)
- Temporal sequence validated (consistent ordering in >80% trajectories)
- Cross-domain replication (all three domains show orchestration)
- **Result**: TNFR Force Dynamics Framework achieves CANONICAL status

**Realistic Case**:
- Composite metric |corr| = 0.84 (marginal improvement)
- Regime classification AUC = 0.92 (good but not exceptional)
- Temporal sequence noisy (50-70% consistency)
- Cross-domain partial (2/3 domains, weaker correlations)
- **Result**: Integrated framework useful but remains RESEARCH

**Worst Case**:
- Fields mostly redundant (PCA: 95% variance in PC1 = Φ_s)
- No regime structure (smooth gradients, no distinct zones)
- Temporal order stochastic (no consistent sequence)
- Cross-domain fails (fields not identifiable or uncorrelated)
- **Result**: Individual fields remain telemetry, no integration benefit

## Success Metrics

- [ ] Correlation matrix + partial correlations documented
- [ ] Composite metric with |corr| ≥ 0.84
- [ ] Force regime phase diagram with ≥3 zones
- [ ] Temporal sequence ordering in ≥60% of trajectories
- [ ] Operator-field signature matrix (13 × 4) with ≥8 significant
- [ ] Cross-domain orchestration validated in ≥2 domains

## Deliverables

1. **Analysis Report**: `docs/research/INTEGRATED_FORCE_DYNAMICS.md`
2. **Code**: Composite metrics + regime classifier in `src/tnfr/physics/forces.py`
3. **Tests**: `tests/physics/test_force_orchestration.py`
4. **Data**: `benchmarks/results/integrated_forces_*.jsonl`
5. **Visualization**: Force regime phase diagram (intensity vs topology heatmap)

---

**Estimated Effort**: 7-10 days (comprehensive analysis + cross-domain validation)
**Priority**: High (synthesizes all individual field studies)
**Dependencies**: Issues #2926 (|∇φ|), #2927 (K_φ), #2928 (ξ_C)
**Assigned to**: @copilot (automated research)
